shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -136,13 +136,13 @@ trait TypeLikeImpl extends Located { self: TypeLike =>
       // })).mkString("\n")
       // })).mkString("", "\n", "\n")
       })).mkString
-    case NuTypeDef(kind @ Als, nme, tparams, params, ctor, plain, sig, parents, sup, ths, body) =>
+    case NuTypeDef(kind @ Als, nme, tparams, params, ctor, sig, parents, sup, ths, body) =>
       s"type ${nme.name}${tparams.map(_._2.showIn(ctx, 0)).mkStringOr(", ", "[", "]")} = ${
         sig.getOrElse(die).showIn(ctx, 0)}"
-    case NuTypeDef(kind, nme, tparams, params, ctor, plain, sig, parents, sup, ths, body) =>
+    case NuTypeDef(kind, nme, tparams, params, ctor, sig, parents, sup, ths, body) =>
       val bodyCtx = ctx.indent
       s"${kind.str} ${nme.name}${tparams.map(_._2.showIn(ctx, 0)).mkStringOr(", ", "[", "]")}(${
```

Comment:
Let's refactor this bit so that a class with no parameter lists actually is _not_ printed as having an empty parameter list!

Reply:
added a pattern matching to check if parameter lists are none. should I also apply this on the `compiler` subproject?

Reply:
What do you mean?

Reply:
in `compiler/shared/main/scala/mlscript/compiler/PrettyPrinter.scala`:
```scala
def showTypeDef(tyDef: NuTypeDef, indent: Int = 0): String =
  s"${tyDef.kind.str} ${tyDef.nme.name}"
    + (if tyDef.tparams.isEmpty
       then ""
       else tyDef.tparams.map(_._2.name).mkString("[", ",", "]"))
    + "(" + tyDef.params.getOrElse(Tup(Nil)) + ")"
    + (if tyDef.parents.isEmpty
       then ""
       else ": " + tyDef.parents.map(_.toString).mkString(", "))
    + showTypingUnit(tyDef.body, indent + 1)
```
Should we also check if `tyDef.params` is none?

Reply:
Sure, but I wasn't aware that the type definition pretty-printing functionality was duplicated in this sub-project (probably because it predates the implementation in the main project). Do you think we could just use the main project implementation here?

Reply:
Never mind, I just noticed that this is for debugging.

https://github.com/hkust-taco/mlscript/blob/703200f0cf9f16f654a85df326e8a32bb7e2728f/compiler/shared/main/scala/mlscript/compiler/PrettyPrinter.scala#L6

So it's probably fine to just update it as you described and otherwise leave it alone!

Reply:
Updated.